usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ParseSingleRowForFileTest.java

```diff
@@ -21,52 +21,69 @@
-import org.mobilitydata.gtfsvalidator.usecase.notice.base.InfoNotice;
 import org.mobilitydata.gtfsvalidator.usecase.notice.base.Notice;
-import org.mobilitydata.gtfsvalidator.usecase.notice.base.WarningNotice;
 import org.mobilitydata.gtfsvalidator.usecase.notice.error.CannotConstructDataProviderNotice;
 import org.mobilitydata.gtfsvalidator.usecase.port.GtfsSpecRepository;
 import org.mobilitydata.gtfsvalidator.usecase.port.RawFileRepository;
 import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
 
 import java.util.*;
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.core.IsInstanceOf.instanceOf;
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
 class ParseSingleRowForFileTest {
 
+    @Mock
+    List<Notice> noticeList = new ArrayList<>();
+    @InjectMocks
+    ValidationResultRepository mockResultRepo;
+
+    @Mock
+    EntityParser mockEntityParser = new EntityParser(Collections.emptyList());
+    @InjectMocks
+    GtfsSpecRepository mockSpecRepo;
+
     @Test
     void shouldValidateAndParseOneByOne() {
 
-        MockResultRepo resultRepo = new MockResultRepo();
-        MockSpecRepo specRepo = new MockSpecRepo();
+        ValidationResultRepository mockResultRepo = mockResultRepository();
+        GtfsSpecRepository mockSpecRepo = mockSpecRepository();
+        RawFileRepository mockFileRepo = mockFileRepository();
 
         ParseSingleRowForFile underTest = new ParseSingleRowForFile(
                 RawFileInfo.builder().filename("test.tst").build(),
-                new MockRawFileRepo(),
-                specRepo,
-                resultRepo
+                mockFileRepo,
+                mockSpecRepo,
+                mockResultRepo
         );
 
         assertTrue(underTest.hasNext());
         underTest.execute();
-        assertEquals(0, resultRepo.noticeList.size());
-        assertEquals(1, specRepo.parser.callToValidateNumericTypesCount);
-        assertEquals(1, specRepo.parser.callToParseCount);
+        assertEquals(0, noticeList.size());
+        assertEquals(1, mockEntityParser.callToValidateNumericTypesCount);
+        assertEquals(1, mockEntityParser.callToParseCount);
 
         assertTrue(underTest.hasNext());
         underTest.execute();
-        assertEquals(0, resultRepo.noticeList.size());
-        assertEquals(2, specRepo.parser.callToValidateNumericTypesCount);
-        assertEquals(2, specRepo.parser.callToParseCount);
+        assertEquals(0, noticeList.size());
+        assertEquals(2, mockEntityParser.callToValidateNumericTypesCount);
+        assertEquals(2, mockEntityParser.callToParseCount);
```

Comment:
Not sure where those two spy methods come from

Reply:
See them defined at the end of the file

Reply:
The mocked repo is using an injected object as attribute at the begin of class. We could add inline comments to clarify.